packages/app/src/composables/useCloudSpecData.ts

```diff
@@ -0,0 +1,119 @@
+import { gql, useMutation } from '@urql/vue'
+import { Ref, computed, watch } from 'vue'
+import { useDebounce } from '@vueuse/core'
+import { CloudData_RefetchDocument, SpecsListFragment } from '../generated/graphql'
+
+gql`
+mutation CloudData_Refetch ($ids: [ID!]!) {
+  loadRemoteFetchables(ids: $ids){
+    id
+    fetchingStatus
+  }
+}
+`
+
+type NonNullCloudSpec = Exclude<SpecsListFragment['cloudSpec'], undefined | null>
+
+/**
+ * Set up watchers & subscriptions to clear/refetch cloud data based on project state relative to latest data in the cloud
+ * @param isProjectDisconnected Whether project has a valid connection to the cloud
+ * @param isOffline  Whether app is currently offline (network state)
+ * @param projectId Cloud project ID (slug)
+ * @param mostRecentUpdate Date/time of the latest project update in the cloud
+ * @param displayedSpecs Spec entries currently displayed in the app
+ * @param allSpecs All project spec entries
+ * @returns Trigger functions
+ */
+export function useCloudSpecData (
+  isProjectDisconnected: Ref<boolean>,
+  isOffline: Ref<boolean>,
+  projectId: string | null | undefined,
+  mostRecentUpdate: Ref<string | null>,
+  displayedSpecs: Ref<(SpecsListFragment | undefined)[]>,
+  allSpecs: (SpecsListFragment | undefined)[],
+) {
+  const refetchMutation = useMutation(CloudData_RefetchDocument)
+
+  const isCloudSpecOlderThan = (item: NonNullCloudSpec, comparisonDttm: string | null) => {
+    if (item.data?.__typename !== 'CloudProjectSpec' && item.data?.__typename !== 'CloudProjectSpecNotFound') {
```

Comment:
We also have `dayjs` in the dependencies, not sure if we need it for this particular comparison, but wanted to note this for the future if we need anything more complex w.r.t day/time.